### PR TITLE
Patch/size validator fix

### PR DESCRIPTION
**Pull request by KrisKnabel**
Noticed there were console errors in next for the size prop validator.  Created a more robust prop validator function in the size mixin.  This branch contains the patch/icon-inherit-color changes, so this branch should be merged AFTER that branch.

### CODE REVIEW

**Comment by lancetharper** [comment]
I fixed console errors on the reset branch I'm working on with the propValidator function we have in utils if you want to just cherrypick this into yours for now 

a80e232713bd8d9e1c39923177dff0903d959eef

**Comment by KrisKnabel** [comment]
I don't think this is a fix that should be part of the validator mixin, but I could be wrong.  I'll look at what you did and see if that makes more sense.

**Comment by KrisKnabel** [comment]
So Lance had a more elegant solution using the prop-validator utility he created.  I copied/cherrypicked his solution from 

https://github.com/rei/rei-cedar/commit/a80e232713bd8d9e1c39923177dff0903d959eef

**Comment by KrisKnabel** [comment]
This is ready for PR Review

File: src/mixins/size.js [hunk 1]
```diff
@@ -23,7 +22,10 @@ export default {
     size: {
       type: String,
       default: 'medium',
-      validator: value => (sizes.indexOf(value) >= 0) || false,
+      validator: value => propValidator(
+        value,
+        ['small', 'medium', 'large'],
```
Comment (zfolwick) [inline src/mixins/size.js:27]: @mhewson - didn't Nathan Curtis specifically warn us against using "small/medium/large"?

Comment (zfolwick) [reply src/mixins/size.js:27]: should there be an "extra-small"?

Comment (KrisKnabel) [reply src/mixins/size.js:27]: there is an extra-small(xs) breakpoint, but we have decided that there is only `small, medium, large ` for sizes.  If someone REALLY REALLY wanted to have an additional size such as `Extra Small` or `Extra Large`, they could use the `modifier` mixin for that.

**Review by lancetharper — APPROVED** [review]

**Pull request merged**